sled-agent/src/services.rs

```diff
@@ -1204,14 +1211,59 @@ impl ServiceManager {
         Ok(())
     }
 
+    pub fn boottime_rewrite(&self, zones: &Vec<RunningZone>) {
+        if self
+            .inner
+            .time_synced
+            .compare_exchange(false, true, Ordering::Acquire, Ordering::Relaxed)
+            .is_err()
+        {
+            // Already done.
+            return;
+        }
+
+        let now = SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .expect("SystemTime before UNIX EPOCH");
+
+        info!(self.inner.log, "Setting boot time to {:?}", now);
+
+        let files: Vec<String> = zones
+            .iter()
+            .map(|z| z.root())
+            .chain(iter::once("".to_string()))
```

Comment:
This is for the database of the global zone, right?

When we create zones in the future, zones will derive values from the GZ's notion of time, right?

Reply:
Yes, and yes. There's only one system clock and once it's right then new zones will be fine, and will create their own `wtmpx` and `utmpx` files with correct timestamps. We just have to take a pass through the databases in the GZ and in any running zones, once time is synchronised for the first time.

Reply:
Makes sense, good catch to fix-up the old zones in place.